### PR TITLE
Support $min field update operator

### LINKED ISSUE #623: Support `$min` field update operator
We already have `$max` update operator. Let's implement `$min`: https://www.mongodb.com/docs/manual/reference/operator/update/min/

We should check if the existing `types.Array.Min` and `types.Array.Max` methods are being used and correct. If they are not, we should fix them or remove them.

If, during implementation, we encounter some minor apparent bugs with the existing `$max` operator, we should fix them. However, we should create a new issue if they are not small or clear.

Unlike usual, we should not do other spot refactoring in this issue. Instead, please add comments to this issue, noting all refactoring opportunities.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #623.

For this task, issue #623 specifies to check usage of `array.Max()` and `array.Min()`. They are used by `sort()` in #1646.
## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by chilagrow** [comment]
The failing linter will be happy when *Remove Incomparable* #1646 is merged. 

```
Error: internal/handlers/common/update.go:396:4: missing cases in switch of type types.CompareResult: Incomparable (exhaustive)
			switch res {
			^
task: Failed to run task "lint": exit status 1
Error: Process completed with exit code 1.
```

File: internal/handlers/common/update.go [hunk 1]
```diff
@@ -368,6 +375,43 @@ func processMaxFieldExpression(doc *types.Document, updateV any) (bool, error) {
 	return changed, nil
 }
 
+// processMinFieldExpression changes document according to $min operator.
+// If the document was changed it returns true.
+func processMinFieldExpression(doc *types.Document, updateV any) (bool, error) {
+	minExpression := updateV.(*types.Document)
+	minExpression = minExpression.SortFieldsByKey()
```
Comment (chilagrow) [inline internal/handlers/common/update.go:382]: Sorting _update_ document fields by key to ensure _MultipleQueriesSorted_ test passes.

**Review by w84thesun — APPROVED** [review]
LGTM!

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by w84thesun — APPROVED** [review]

**Review by Fashander — APPROVED** [review]

**Pull request merged**